navigation/create_control/include/create_hw_interface.hpp

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include "create/create.h"
+#include <controller_manager/controller_manager.h>
+#include <hardware_interface/joint_command_interface.h>
+#include <hardware_interface/joint_state_interface.h>
+#include <hardware_interface/robot_hw.h>
+#include <ros/ros.h>
+
+class Create2_interface : public hardware_interface::RobotHW {
+public:
+  Create2_interface() : model_(create::RobotModel::CREATE_2) {
+    // connect and register the joint state interface
+    hardware_interface::JointStateHandle state_handle_a(
+        "left_wheel_joint", &pos[0], &vel[0], &eff[0]);
+    jnt_state_interface.registerHandle(state_handle_a);
+
+    hardware_interface::JointStateHandle state_handle_b(
+        "right_wheel_joint", &pos[1], &vel[1], &eff[1]);
+    jnt_state_interface.registerHandle(state_handle_b);
+
+    registerInterface(&jnt_state_interface);
+
+    // connect and register the joint velocity interface
+    hardware_interface::JointHandle vel_handle_left(
+        jnt_state_interface.getHandle("left_wheel_joint"), &cmd[0]);
+    jnt_vel_interface.registerHandle(vel_handle_left);
+
+    hardware_interface::JointHandle vel_handle_right(
+        jnt_state_interface.getHandle("right_wheel_joint"), &cmd[1]);
+    jnt_vel_interface.registerHandle(vel_handle_right);
+
+    registerInterface(&jnt_vel_interface);
+    model_ = create::RobotModel::CREATE_2;
+    robot_.reset(new create::Create(model_));
+
+    if (!robot_->connect("/dev/ttyUSB0",
```

Comment:
You can get these parameters from [here](https://github.com/RoboticaUtnFrba/create_autonomy/blob/kinetic-devel/ca_node/launch/create_2.launch#L10).

Reply:
Load the parameters from the YAML file.

Reply:
This is still pending

Reply:
I'll wait for this change to test it again.

Reply:
Addressed